### PR TITLE
Task-1 Create search, count and genre list components

### PR DESCRIPTION
**Pull request by PetkevichVladislav**
Components created according to the task: 
⚛️ Counter

A component that renders a numeric value and two buttons: one to decrement the value by 1, another to increment the value by 1.
The component should take an initial value in a property.
The component should be written using EcmaScript classes syntax, the `render` method should use `React.createElement` API (without JSX).

⚛️ SearchForm

Refer to the [design prototype](https://www.figma.com/file/fKGjrOqR6nJe6LYJopGCZ8/CDP-Home-Task-%E2%80%93-React-v1), implement a component that renders a search input and a button that triggers a new search.
The component should accept two properties:

Initial search query. Use the value to set the initial value for the input
A "onSearch" callback property. Call the callback property every time the user presses Enter when the input has focus or when the user clicks the Search button. Pass current input value in callback arguments.
⚛️ GenreSelect

Refer to the [design prototype](https://www.figma.com/file/fKGjrOqR6nJe6LYJopGCZ8/%5BCDP%5D-Home-Task-%E2%80%93-React-v1?node-id=0%3A1), implement a component that renders a list of movie genres with currently selected genre highlighted.
The component should accept three properties:

A list of genre names to display. Use the incoming list to render genre buttons.
A name of currently selected genre. Use the name to identify which button to highlight.
A "onSelect" callback property. Call the callback function when the user clicks on any genre button. Pass respective genre name to the callback arguments.

The counter component contains two variants:
1. Created as react class according to the task.
2. (Commented) Created as a function for the furure refactoring.

### CODE REVIEW

File: src/components/Counter/Counter.tsx [hunk 1]
```diff
@@ -1,17 +1,89 @@
...
+// }
+
+// Code to follow task restrictions.
+interface ICounterProps {
+    initialValue: number;
+  }
+  
+  interface ICounterState {
+    count: number;
+  }
```
Comment (iZamza) [inline src/components/Counter/Counter.tsx]: please use a separate file for types

File: src/components/Header/Header.tsx [hunk 2]
```diff
@@ -1,10 +1,17 @@
...
+
 import Logo from "../Logo/Logo";
 import Search from "../Search/Search";
 
-import "./Header.scss"
+function Header() {
+    const [searchQuery, setSearchQuery] = useState<string>('Comedy');
+
+    const searchByQuery = useCallback((query: string) => {
+        console.log(query);
```
Comment (iZamza) [inline src/components/Header/Header.tsx:12]: probably we should create some functionality here.

File: src/components/List/List.scss [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+                &::after{
+                    content: "";
+                    position: absolute;
+                    bottom: -23px;
+                    width: 100%;
+                    height: 3px;
+                    left: 0px;
+                    background-color: #F65261;
+                }
+            }
```
Comment (iZamza) [inline src/components/List/List.scss:26]: spaces / tabulation

File: src/components/List/List.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+interface IGenreListProps {
+    itemNames: string[],
+    preselectedItemName: string,
+    onSelect: (itemName: string) => void,
+}
+
+const List: FC<IGenreListProps> = ({itemNames, preselectedItemName, onSelect}) => {
+    const [selectedItemName, setSelectedItemName] = useState(preselectedItemName);
+    const listItems = itemNames.map((itemName) => {
+        var className = itemName === selectedItemName ? selectedItemClassName : unselectedItemClassName;
```
Comment (iZamza) [inline src/components/List/List.tsx]: const

File: src/components/List/List.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+const unselectedItemClassName = "list__item";
+
+interface IGenreListProps {
+    itemNames: string[],
+    preselectedItemName: string,
+    onSelect: (itemName: string) => void,
+}
+
+const List: FC<IGenreListProps> = ({itemNames, preselectedItemName, onSelect}) => {
+    const [selectedItemName, setSelectedItemName] = useState(preselectedItemName);
```
Comment (iZamza) [inline src/components/List/List.tsx]: please add types

File: src/components/List/List.tsx [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    const listItems = itemNames.map((itemName) => {
+        var className = itemName === selectedItemName ? selectedItemClassName : unselectedItemClassName;
+        return <li key={itemName} className={className} onClick={() => SelectItem(itemName)}>{itemName}</li>;
+    });
+
+    const SelectItem = (itemName: string) => {
```
Comment (iZamza) [inline src/components/List/List.tsx]: `selectItem`

**Review by iZamza — APPROVED** [review]

File: src/components/List/List.tsx (hunk 6 above)
Comment (PetkevichVladislav) [reply src/components/List/List.tsx]: Renamed

File: src/components/List/List.tsx (hunk 5 above)
Comment (PetkevichVladislav) [reply src/components/List/List.tsx]: Type added, thank you for the comment.

File: src/components/List/List.tsx (hunk 4 above)
Comment (PetkevichVladislav) [reply src/components/List/List.tsx]: var replaced to const, thank you for the comment.

File: src/components/List/List.scss (hunk 3 above)
Comment (PetkevichVladislav) [reply src/components/List/List.scss:26]: Tabulation fixed, thank you for the comment.

File: src/components/Header/Header.tsx (hunk 2 above)
Comment (PetkevichVladislav) [reply src/components/Header/Header.tsx:12]: We will add functionality in future tasks.

File: src/components/Counter/Counter.tsx (hunk 1 above)
Comment (PetkevichVladislav) [reply src/components/Counter/Counter.tsx]: Moved to separate file, thank you for the comment.

**Pull request merged**